File: Converted_POS_08May2025/Converted_POS/Models/clsProfileMaster.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
 
using System.Linq;
using System.Threading.Tasks;

namespace Converted_POS.Models
{
    public class clsProfileMaster
    {
        public int cmp_id { get; set; }
        public int? profile_id { get; set; }
        [Required]
        public string profile_name { get; set; }
        public decimal? bonus_point { get; set; }
        public decimal? amount { get; set; }
        public DateTime? created_date { get; set; }
        public DateTime? modify_date { get; set; }
        public decimal? purchase_amount { get; set; }
        public decimal? earn_bonus { get; set; }
        public String Tran_Type { get; set; }
        public decimal? discount_percent { get; set; }
        public bool IsDefault { get; set; }
    }

    //public class clsProfileMaster_DAL
    //{
        //To View all details    
        //public IEnumerable<clsProfileMaster> SelectAll(int? c_id, string conn)
        //{
        //    List<clsProfileMaster> list = new List<clsProfileMaster>();

        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("Get_M_Profile", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", c_id);

        //        con.Open();
        //        SqlDataReader rdr = cmd.ExecuteReader();

        //        while (rdr.Read())
        //        {
        //            string active = "";
        //            clsProfileMaster DType = new clsProfileMaster();

        //            DType.profile_id = Convert.ToInt32(rdr["profile_id"]);
        //            DType.profile_name = rdr["profile_name"].ToString();


        //            list.Add(DType);
        //        }
        //        con.Close();
        //    }
        //    return list;
        //}

        //Get the details of a particular data  
        //public clsProfileMaster Select(int? c_id, int? id, string conn)
        //{
        //    clsProfileMaster profile = new clsProfileMaster();

        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("Get_M_Profile_info", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@profile_id", id);
        //        cmd.Parameters.AddWithValue("@cmp_id", c_id);

        //        con.Open();
        //        SqlDataReader rdr = cmd.ExecuteReader();

        //        while (rdr.Read())
        //        {
        //            profile.profile_id = Convert.ToInt32(rdr["profile_id"]);
        //            profile.profile_name = rdr["profile_name"].ToString();
        //            profile.bonus_point = Convert.ToDecimal(rdr["bonus_point"]);
        //            profile.amount = Convert.ToDecimal(rdr["amount"]);
        //            profile.discount_percent = Convert.ToDecimal(rdr["discount_percent"]);
        //            profile.earn_bonus = Convert.ToDecimal(rdr["earn_bonus"]);
        //            profile.purchase_amount = Convert.ToDecimal(rdr["purchase_amount"]);

        //            if (rdr["Is_Default"].ToString() == "1")
        //            {
        //                profile.IsDefault = true;
        //            }
        //            else
        //            {
        //                profile.IsDefault = false;
        //            }

        //        }

        //        con.Close();
        //    }
        //    return profile;
        //}


        //To Add new record    
        //public void Insert(clsProfileMaster profile, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Profile", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", profile.cmp_id);
        //        cmd.Parameters.AddWithValue("@profile_id", profile.profile_id);
        //        cmd.Parameters.AddWithValue("@profile_name", profile.profile_name);
        //        cmd.Parameters.AddWithValue("@bonus_point", profile.bonus_point);
        //        cmd.Parameters.AddWithValue("@amount", profile.amount);
        //        cmd.Parameters.AddWithValue("@purchase_amount", profile.purchase_amount);
        //        cmd.Parameters.AddWithValue("@earn_bonus", profile.earn_bonus);
        //        cmd.Parameters.AddWithValue("@discount_percent", profile.discount_percent);
        //        cmd.Parameters.AddWithValue("@IsDefaul", profile.IsDefault);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "I");

        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}


        //To Update record    
        //public void Update(clsProfileMaster profile, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Profile", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", profile.cmp_id);
        //        cmd.Parameters.AddWithValue("@profile_id", profile.profile_id);
        //        cmd.Parameters.AddWithValue("@profile_name", profile.profile_name);
        //        cmd.Parameters.AddWithValue("@bonus_point", profile.bonus_point);
        //        cmd.Parameters.AddWithValue("@amount", profile.amount);
        //        cmd.Parameters.AddWithValue("@purchase_amount", profile.purchase_amount);
        //        cmd.Parameters.AddWithValue("@earn_bonus", profile.earn_bonus);
        //        cmd.Parameters.AddWithValue("@discount_percent", profile.discount_percent);
        //        cmd.Parameters.AddWithValue("@IsDefaul", profile.IsDefault);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "U");

        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}

        //Delete record    
        //public void Delete(clsProfileMaster profile, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Profile", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", profile.cmp_id);
        //        cmd.Parameters.AddWithValue("@profile_id", profile.profile_id);
        //        cmd.Parameters.AddWithValue("@profile_name", profile.profile_name);
        //        cmd.Parameters.AddWithValue("@bonus_point", profile.bonus_point);
        //        cmd.Parameters.AddWithValue("@amount", profile.amount);
        //        cmd.Parameters.AddWithValue("@purchase_amount", profile.purchase_amount);
        //        cmd.Parameters.AddWithValue("@earn_bonus", profile.earn_bonus);
        //        cmd.Parameters.AddWithValue("@discount_percent", profile.discount_percent);
        //        cmd.Parameters.AddWithValue("@IsDefaul", profile.IsDefault);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "D");

        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}


    //}
}

File: POS_10Apr2025/POS/App_Code/clsProfileMaster.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data


Public Class Controller_clsProfileMaster

    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
    Public Sub New()
    End Sub
#Region "Public Variables"

    Private _cmp_id As Integer
    Private _profile_id As Integer
    Private _profile_name As String
    Private _bonus_point As Integer
    Private _amount As Integer
    Private _created_date As System.DateTime
    Private _modify_date As System.DateTime
    Private _purchase_amount As Integer
    Private _earn_bonus As Integer
    Private _Tran_Type As String
    Private _discount_percent As Integer
    Private _IsDefaul As Integer
    Private objclsProfileMaster As clsProfileMaster
#End Region
#Region "Public Property"

    Public Property IsDefaul() As Integer
        Get
            Return _IsDefaul
        End Get
        Set(ByVal value As Integer)
            _IsDefaul = value
        End Set
    End Property


    Public Property cmp_id() As Integer
        Get
            Return _cmp_id
        End Get
        Set(ByVal value As Integer)
            _cmp_id = value
        End Set
    End Property


    Public Property profile_id() As Integer
        Get
            Return _profile_id
        End Get
        Set(ByVal value As Integer)
            _profile_id = value
        End Set
    End Property

    Public Property profile_name() As String
        Get
            Return _profile_name
        End Get
        Set(ByVal value As String)
            _profile_name = value
        End Set
    End Property

    Public Property bonus_point() As Integer
        Get
            Return _bonus_point
        End Get
        Set(ByVal value As Integer)
            _bonus_point = value
        End Set
    End Property

    Public Property amount() As Integer
        Get
            Return _amount
        End Get
        Set(ByVal value As Integer)
            _amount = value
        End Set
    End Property

    Public Property created_date() As System.DateTime
        Get
            Return _created_date
        End Get
        Set(ByVal value As System.DateTime)
            _created_date = value
        End Set
    End Property
    Public Property modify_date() As System.DateTime
        Get
            Return _modify_date
        End Get
        Set(ByVal value As System.DateTime)
            _modify_date = value
        End Set
    End Property
    Public Property purchase_amount() As Integer
        Get
            Return _purchase_amount
        End Get
        Set(ByVal value As Integer)
            _purchase_amount = value
        End Set
    End Property
    Public Property earn_bonus() As Integer
        Get
            Return _earn_bonus
        End Get
        Set(ByVal value As Integer)
            _earn_bonus = value
        End Set
    End Property
    Public Property Tran_Type() As String
        Get
            Return _Tran_Type
        End Get
        Set(ByVal value As String)
            _Tran_Type = value
        End Set
    End Property
    Public Property discount_percent() As Integer
        Get
            Return _discount_percent
        End Get
        Set(ByVal value As Integer)
            _discount_percent = value
        End Set
    End Property
#End Region
    Public Function Insert() As Boolean
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.cmp_id = cmp_id
            objclsProfileMaster.profile_id = profile_id
            objclsProfileMaster.profile_name = profile_name
            objclsProfileMaster.bonus_point = bonus_point
            objclsProfileMaster.purchase_amount = purchase_amount
            objclsProfileMaster.earn_bonus = earn_bonus
            objclsProfileMaster.amount = amount
            objclsProfileMaster.discount_percent = discount_percent
            objclsProfileMaster.IsDefaul = IsDefaul
            If objclsProfileMaster.Insert() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [SelectAll]() As DataTable
        Dim dt As DataTable
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.cmp_id = cmp_id
            dt = objclsProfileMaster.[SelectAll]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Delete() As Boolean
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.cmp_id = cmp_id
            objclsProfileMaster.profile_id = profile_id
            objclsProfileMaster.profile_name = profile_name
            objclsProfileMaster.bonus_point = bonus_point
            objclsProfileMaster.amount = amount
            objclsProfileMaster.purchase_amount = purchase_amount
            objclsProfileMaster.earn_bonus = earn_bonus
            objclsProfileMaster.Tran_Type = Tran_Type
            objclsProfileMaster.discount_percent = discount_percent
            objclsProfileMaster.IsDefaul = IsDefaul
            If objclsProfileMaster.Delete() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Update() As Boolean
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.cmp_id = cmp_id
            objclsProfileMaster.profile_id = profile_id
            objclsProfileMaster.profile_name = profile_name
            objclsProfileMaster.bonus_point = bonus_point
            objclsProfileMaster.amount = amount
            objclsProfileMaster.purchase_amount = purchase_amount
            objclsProfileMaster.earn_bonus = earn_bonus
            objclsProfileMaster.Tran_Type = Tran_Type
            objclsProfileMaster.discount_percent = discount_percent
            objclsProfileMaster.IsDefaul = IsDefaul
            If objclsProfileMaster.Update() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [Select]() As DataTable
        Dim dt As DataTable
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.profile_id = profile_id
            objclsProfileMaster.cmp_id = cmp_id
            dt = objclsProfileMaster.[Select]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function

    Public Function SelectProfile() As DataTable
        Dim dt As DataTable
        Try
            objclsProfileMaster = New clsProfileMaster()
            objclsProfileMaster.cmp_id = cmp_id
            dt = objclsProfileMaster.SelectProfile()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function

End Class


Public Class clsProfileMaster
    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
#Region "Public Variables"

    Private _cmp_id As Integer
    Private _profile_id As Integer
    Private _profile_name As String
    Private _bonus_point As Integer
    Private _amount As Integer
    Private _created_date As System.DateTime
    Private _modify_date As System.DateTime
    Private _purchase_amount As Integer
    Private _earn_bonus As Integer
    Private _Tran_Type As String
    Private _discount_percent As Integer
    Private _IsDefaul As Integer
    Private objclsProfileMaster As clsProfileMaster
#End Region
#Region "Public Property"

    Public Property IsDefaul() As Integer
        Get
            Return _IsDefaul
        End Get
        Set(ByVal value As Integer)
            _IsDefaul = value
        End Set
    End Property
    Public Property cmp_id() As Integer
        Get
            Return _cmp_id
        End Get
        Set(ByVal value As Integer)
            _cmp_id = value
        End Set
    End Property
    Public Property profile_id() As Integer
        Get
            Return _profile_id
        End Get
        Set(ByVal value As Integer)
            _profile_id = value
        End Set
    End Property

    Public Property profile_name() As String
        Get
            Return _profile_name
        End Get
        Set(ByVal value As String)
            _profile_name = value
        End Set
    End Property

    Public Property bonus_point() As Integer
        Get
            Return _bonus_point
        End Get
        Set(ByVal value As Integer)
            _bonus_point = value
        End Set
    End Property

    Public Property amount() As Integer
        Get
            Return _amount
        End Get
        Set(ByVal value As Integer)
            _amount = value
        End Set
    End Property

    Public Property created_date() As System.DateTime
        Get
            Return _created_date
        End Get
        Set(ByVal value As System.DateTime)
            _created_date = value
        End Set
    End Property
    Public Property modify_date() As System.DateTime
        Get
            Return _modify_date
        End Get
        Set(ByVal value As System.DateTime)
            _modify_date = value
        End Set
    End Property
    Public Property purchase_amount() As Integer
        Get
            Return _purchase_amount
        End Get
        Set(ByVal value As Integer)
            _purchase_amount = value
        End Set
    End Property
    Public Property earn_bonus() As Integer
        Get
            Return _earn_bonus
        End Get
        Set(ByVal value As Integer)
            _earn_bonus = value
        End Set
    End Property

    Public Property Tran_Type() As String
        Get
            Return _Tran_Type
        End Get
        Set(ByVal value As String)
            _Tran_Type = value
        End Set
    End Property
    Public Property discount_percent() As Integer
        Get
            Return _discount_percent
        End Get
        Set(ByVal value As Integer)
            _discount_percent = value
        End Set
    End Property
#End Region
    Public Function Insert() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_id", profile_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_name", profile_name)
            oColSqlparram.Add("@bonus_point", bonus_point, SqlDbType.Int)
            oColSqlparram.Add("@amount", amount, SqlDbType.Int)
            oColSqlparram.Add("@purchase_amount", purchase_amount, SqlDbType.Int)
            oColSqlparram.Add("@earn_bonus", earn_bonus, SqlDbType.Int)
            oColSqlparram.Add("@discount_percent", discount_percent, SqlDbType.Int)
            oColSqlparram.Add("@IsDefaul", IsDefaul, SqlDbType.Int)
            oColSqlparram.Add("@Tran_Type", "I")
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Profile", oColSqlparram)
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [SelectAll]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Profile", oColSqlparram)

        Return dtlogin
    End Function

    Public Function [Select]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@profile_id", profile_id, SqlDbType.Int)
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Profile_info", oColSqlparram)

        Return dtlogin
    End Function


    Public Function Delete() As Boolean
        Try

            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_id", profile_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_name", profile_name)
            oColSqlparram.Add("@bonus_point", bonus_point, SqlDbType.Int)
            oColSqlparram.Add("@amount", amount, SqlDbType.Int)
            oColSqlparram.Add("@purchase_amount", purchase_amount, SqlDbType.Int)
            oColSqlparram.Add("@earn_bonus", earn_bonus, SqlDbType.Int)
            oColSqlparram.Add("@discount_percent", discount_percent, SqlDbType.Int)
            oColSqlparram.Add("@IsDefaul", IsDefaul, SqlDbType.Int)
            oColSqlparram.Add("@Tran_Type", "D")
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Profile", oColSqlparram)
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Update() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_id", profile_id, SqlDbType.Int)
            oColSqlparram.Add("@profile_name", profile_name)
            oColSqlparram.Add("@bonus_point", bonus_point, SqlDbType.Int)
            oColSqlparram.Add("@amount", amount, SqlDbType.Int)
            oColSqlparram.Add("@purchase_amount", purchase_amount, SqlDbType.Int)
            oColSqlparram.Add("@earn_bonus", earn_bonus, SqlDbType.Int)
            oColSqlparram.Add("@discount_percent", discount_percent, SqlDbType.Int)
            oColSqlparram.Add("@IsDefaul", IsDefaul, SqlDbType.Int)
            oColSqlparram.Add("@Tran_Type", "U")
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Profile", oColSqlparram)
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function


    Public Function SelectProfile() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Profile", oColSqlparram)

        Return dtlogin
    End Function
End Class
